go/vt/vtadmin/api.go

```diff
@@ -555,3 +602,152 @@ func (api *API) getClustersForRequest(ids []string) ([]*cluster.Cluster, []strin
 
 	return clusters, ids
 }
+
+// VTExplain is part of the vtadminpb.VTAdminServer interface.
+func (api *API) VTExplain(ctx context.Context, req *vtadminpb.VTExplainRequest) (*vtadminpb.VTExplainResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "API.VTExplain")
+	defer span.Finish()
+
+	if req.Cluster == "" {
+		return nil, fmt.Errorf("%w: cluster ID is required", ErrInvalidRequest)
+	}
+
+	if req.Keyspace == "" {
+		return nil, fmt.Errorf("%w: keyspace name is required", ErrInvalidRequest)
+	}
+
+	if req.Sql == "" {
+		return nil, fmt.Errorf("%w: SQL query is required", ErrInvalidRequest)
+	}
+
+	c, ok := api.clusterMap[req.Cluster]
+	if !ok {
+		return nil, ErrUnsupportedCluster
+	}
+
+	tablet, err := api.findTablet(ctx, c, func(t *vtadminpb.Tablet) bool {
+		return t.Tablet.Keyspace == req.Keyspace && t.Tablet.Type == topodata.TabletType_REPLICA && t.State == vtadminpb.Tablet_SERVING
```

Comment:
In our internal prototype, we only check for replicas. Would other tablet types (like rdonly) be appropriate here?

Reply:
There's `IsInServingGraph` which does what you are looking for.

Reply:
`IsInServingGraph` is perfect, thanks!